### DISCUSSION TITLE
Is newrelic_infra_alert_condition really legacy resource?

### DISCUSSION THREAD

**Discussion by gecko655** [newrelic/terraform-provider-newrelic#2508] [Q&A] [answered]
I'd like to create infrastructure "host not reporting" alert condition, referring these documents: https://docs.newrelic.com/docs/infrastructure/infrastructure-alerts/infrastructure-alert-conditions/create-infrastructure-host-not-reporting-condition/ https://docs.newrelic.com/docs/infrastructure/new-relic-infrastructure/infrastructure-alert-conditions/infrastructure-alerts-add-edit-or-view-host-alert-information/

From the documentation, it seems that this alert condition can be created by `newrelic_infra_alert_condition` terraform resource, but the document https://registry.terraform.io/providers/newrelic/newrelic/latest/docs/resources/infra_alert_condition says:

> This is a legacy resource. The [newrelic_nrql_alert_condition](https://registry.terraform.io/providers/newrelic/newrelic/latest/docs/resources/nrql_alert_condition) resource is preferred for configuring alerts conditions. In most cases feature parity can be achieved with a NRQL query. This condition type may be deprecated in the future.

Is this true? If so, how can I write the following equivalent alert condition via newrelic_nrql_alert_condition?
I don't think there is a schema for detecting this type of condition via NRQL.

```tf
resource "newrelic_infra_alert_condition" "host_not_reporting" {
  policy_id = newrelic_alert_policy.foo.id

  name        = "Host not reporting"
  description = "Critical alert when the host is not reporting"
  type        = "infra_host_not_reporting"
  where       = "(hostname LIKE '%frontend%')"

  critical {
    duration = 5
  }
}
```

**Comment by gecko655** [comment]
related question: https://github.com/newrelic/terraform-provider-newrelic/discussions/2224
I understand the deprecation policy, but still I don't find how to create a equivalent condition to `newrelic_infra_alert_condition` via `newrelic_nrql_alert_condition` at this moment.

**Comment by pranav-new-relic** [comment, marked as answer]
Hi @gecko655, thanks for posting this question. I've checked with the Alerts team here at New Relic, and it looks like there currently isn't a direct alternative to HNR Infra Alert Conditions via NRQL, unfortunately. To work this around, they suggest using Loss of Signal type conditions (though these cannot substitute HNR conditions completely), while development for an NRQL based solution for HNR alert conditions is underway. [This post](https://forum.newrelic.com/s/hubtopic/aAX8W0000008couWAA/host-not-reporting-with-a-nrql-alert) discusses solutions along the lines of this.

I shall let my team know about this so we can take note of such exceptions in lieu of the planned deprecation. Thanks!

**Reply by gecko655** [reply]
Thank you for your reply! (Sorry for my late response.)

>  it looks like there currently isn't a direct alternative to HNR Infra Alert Conditions via NRQL, unfortunately

I know the discussion post you mentioned, but the proposed solution does not consider successful termination via `systemd`, and alerts are sounded even when infra agent is successfully terminated, while HNR conditions can handle it correctly.
I don't think this spec is acceptable to use in my project.

> development for an NRQL based solution for HNR alert conditions is underway.

I'm glad to hear that!
I'd like to use HNR Infra Alert Conditions at this moment, and will migrate it after the functionality to be available.
